packages/synchronize/lib/src/journal.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:synchronize/src/journal_entry.dart';
+
+part 'journal.g.dart';
+
+/// Contains the journal.
+///
+/// Used for detecting changes and new or deleted files.
+@JsonSerializable()
+class SyncJournal {
+  // Note: This must not be const as otherwise the entries are not modifiable when a const set is used!
+  // ignore: public_member_api_docs
+  SyncJournal(
```

Comment:
I don't know how much it makes sense but a journal is probably always created empty.

I therefore suggest adding a named constructor like `SyncJournal.empty()` that initializes the the `entries` with an empty map.

Reply:
How about named parameter with default empty value? IMO that is even cleaner

Reply:
also possible.
But as the set can't be null it will probably have to be something like:
```dart
SyncJournal({
 Set<SyncJournalEntry>? entries, 
}): entries = entries ?? {};
```